### ISSUE TITLE
Image segmentation with VGG16 Backbone

### ISSUE THREAD

**Issue by CHIMECHEFULAMP** [CHIMECHEFULAMP/Housing#3]
import numpy as np
import tensorflow as tf
from tensorflow.keras import layers, models
from tensorflow.keras.applications import VGG16
from tensorflow.keras.optimizers import Adam
import matplotlib.pyplot as plt

# Load the VGG16 model without pre-trained weights and the top layers
base_model = VGG16(weights=None, include_top=False, input_shape=(32, 32, 3))

# Add new layers on top of the base model
x = base_model.output
x = layers.Flatten()(x)  # Flatten the output before passing it to Dense layers
x = layers.Dense(512, activation='relu')(x)
x = layers.Dense(256, activation='relu')(x)
x = layers.Dense(3, activation='softmax')(x)  # Final output layer

# Define the new model
model = models.Model(inputs=base_model.input, outputs=x)

# Print model summary
model.summary()

# Compile the model
optimizer = Adam(learning_rate=0.004)
model.compile(optimizer=optimizer, loss="categorical_crossentropy", metrics=["accuracy"])

# Define dummy datasets for illustration (replace with actual data)
training_set = np.random.rand(100, 32, 32, 3)  # 100 training images
training_labels = tf.keras.utils.to_categorical(np.random.randint(0, 3, 100), 3)
val_set = np.random.rand(20, 32, 32, 3)  # 20 validation images
val_labels = tf.keras.utils.to_categorical(np.random.randint(0, 3, 20), 3)
test_label = np.random.rand(10, 32, 32, 3)  # 10 test images

# Train the model
model.fit(training_set, training_labels, validation_data=(val_set, val_labels), epochs=20, batch_size=16)

# Make predictions
prediction = model.predict(test_label)

# Define a color map for segmentation visualization
color_map = {
    0: [0, 0, 255],  # Blue
    1: [0, 255, 0],  # Green
    2: [255, 0, 0]   # Red
}

# Get segmentation output
segmentation_output_argmax = np.argmax(prediction, axis=-1)

# Create a blank colored output image
coloured_output = np.zeros((*segmentation_output_argmax.shape, 3), dtype=np.uint8)

# Map each class to its color
for label, color in color_map.items():
    coloured_output[segmentation_output_argmax == label] = color

# Display segmentation map
plt.imshow(coloured_output)
plt.axis('off')
plt.show()
